bitcoin/src/crypto/key.rs

```diff
@@ -230,13 +227,53 @@ impl fmt::Display for PublicKey {
     }
 }
 
+/// Error returned while constructing public key from string
+#[derive(Debug, PartialEq)]
+pub enum ParsePubKeyError {
+    /// Error originated while parsing string
+    StringError(FromSliceError),
```

Comment:
In 4717cb9a384afc21817a1fba35573311e4ce2b0b:

`StringError` suggests this is a problem with the string, not with the underlying bytes. I'd call it `Encoding` or something.

Reply:
I'd prefer `Parse` since it's from string. :) Also we don't add the `Error` suffix to variants.

Reply:
It shouldn't be `Hex` because this error variant is only used after hex decoding has succeeded.

Reply:
Oh, yeah, today is hard for my brain. :D It says `FromSlice` not `FromStr`, so yes, it could be `Encoding` but that suggests it's unrelated to the point not being on the curve . Will try to come up with something better.